app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -0,0 +1,96 @@
+import { click, visit } from "@ember/test-helpers";
+import {
+  acceptance,
+  query,
+  updateCurrentUser,
+} from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import { cloneJSON } from "discourse-common/lib/object";
+import TopicFixtures from "discourse/tests/fixtures/topic";
+import I18n from "I18n";
+
+acceptance("User menu", function (needs) {
+  needs.user({ redesigned_user_menu_enabled: true });
+  let requestHeaders = {};
+  needs.pretender((server, helper) => {
+    server.get("/t/1234.json", (request) => {
+      const json = cloneJSON(TopicFixtures["/t/130.json"]);
+      json.id = 1234;
+      json.post_stream.posts.forEach((post) => {
+        post.topic_id = 1234;
+      });
+      requestHeaders = request.requestHeaders;
+      return helper.response(json);
+    });
+  });
+
+  needs.hooks.afterEach(() => {
+    requestHeaders = {};
+  });
+
+  test("clicking on an unread notification", async function (assert) {
+    await visit("/");
+    await click(".d-header-icons .current-user");
+    const unreadNotification = query(".user-menu ul li.replied a");
+    await click(unreadNotification);
+    assert.strictEqual(
+      requestHeaders["Discourse-Clear-Notifications"],
+      123, // id is from the fixtures in fixtures/notification-fixtures.js
+      "the Discourse-Clear-Notifications is set to the notification id in the next ajax request"
```

Comment:
```suggestion
      "the Discourse-Clear-Notifications request header is set to the notification id in the next ajax request"
```

Reply:
I'm slightly unsure what `next ajax request` here means.

Reply:
When you click on an unread notification item, we register the ID of the notification in the our ajax library and then in the next request that our ajax lib makes, we include in the request the `Discourse-Clear-Notifications` header and set it to the notification id.